java/src/test/java/ai/rapids/cudf/DecimalColumnVectorTest.java

```diff
@@ -0,0 +1,255 @@
+
+  private final BigDecimal[] boundaryDecimal64 = new BigDecimal[]{
+      new BigDecimal("999999999999999999"), new BigDecimal("-999999999999999999")};
+
+  private final BigDecimal[] overflowDecimal32 = new BigDecimal[]{
+      BigDecimal.valueOf(Integer.MAX_VALUE), BigDecimal.valueOf(Integer.MIN_VALUE)};
+
+  private final BigDecimal[] overflowDecimal64 = new BigDecimal[]{
+      BigDecimal.valueOf(Long.MAX_VALUE), BigDecimal.valueOf(Long.MIN_VALUE)};
+
+  @BeforeAll
+  public static void setup() {
+    for (int i = 0; i < decimal32Zoo.length; i++) {
+      unscaledDec32Zoo[i] = rdSeed.nextInt() / 10;
+      unscaledDec64Zoo[i] = rdSeed.nextLong() / 10;
+      if (rdSeed.nextBoolean()) {
+        decimal32Zoo[i] = BigDecimal.valueOf(rdSeed.nextInt() / 10, dec32Scale);
+      } else {
+        decimal32Zoo[i] = null;
+      }
+      if (rdSeed.nextBoolean()) {
+        decimal64Zoo[i] = BigDecimal.valueOf(rdSeed.nextLong() / 10, dec64Scale);
+      } else {
+        decimal64Zoo[i] = null;
+      }
+    }
+  }
+
+  @Test
+  public void testCreateColumnVectorBuilder() {
+    try (ColumnVector decColumnVector = ColumnVector.build(DType.create(DType.DTypeEnum.DECIMAL32, -5), 3,
+        (b) -> b.append(BigDecimal.valueOf(123456789, 5)))) {
+      assertFalse(decColumnVector.hasNulls());
+    }
+    try (ColumnVector decColumnVector = ColumnVector.build(DType.create(DType.DTypeEnum.DECIMAL64, -10), 3,
+        (b) -> b.append(BigDecimal.valueOf(1023040506070809L, 10)))) {
+      assertFalse(decColumnVector.hasNulls());
+    }
+  }
+
+  @Test
+  public void testUpperIndexOutOfBoundsException() {
+    try (HostColumnVector decColumnVector = HostColumnVector.fromDecimals(decimal32Zoo)) {
+      assertThrows(AssertionError.class, () -> decColumnVector.getBigDecimal(decimal32Zoo.length));
+    }
+  }
+
+  @Test
+  public void testLowerIndexOutOfBoundsException() {
+    try (HostColumnVector doubleColumnVector = HostColumnVector.fromDecimals(decimal32Zoo)) {
+      assertThrows(AssertionError.class, () -> doubleColumnVector.getBigDecimal(-1));
+    }
+  }
+
+  @Test
+  public void testAddingNullValues() {
+    try (HostColumnVector cv = HostColumnVector.fromDecimals(decimal64Zoo)) {
+      for (int i = 0; i < decimal64Zoo.length; ++i) {
+        assertEquals(decimal64Zoo[i] == null, cv.isNull(i));
+      }
+      assertEquals(Arrays.stream(decimal64Zoo).filter(Objects::isNull).count(), cv.getNullCount());
+    }
+  }
+
+  @Test
+  public void testOverrunningTheBuffer() {
+    try (Builder builder = HostColumnVector.builder(DType.create(DType.DTypeEnum.DECIMAL32, dec32Scale), 3)) {
+      assertThrows(AssertionError.class, () -> builder.appendBoxed(decimal32Zoo).build());
+    }
+    try (Builder builder = HostColumnVector.builder(DType.create(DType.DTypeEnum.DECIMAL64, dec64Scale), 3)) {
+      assertThrows(AssertionError.class, () -> builder.appendUnscaledDecimalArray(unscaledDec64Zoo).build());
+    }
+  }
+
+  @Test
```

Comment:
There should be a unit test for building from BigDecimal values with varying scales and a unit test for appending BigDecimal values with a different scale than used in the existing column vector (if we keep the BigDecimal builders).

Reply:
An additional method `append(BigDecimal value, RoundingMode roundingMode)` was added for `appending BigDecimal values with a different scale`. When roundingMode is null, the scale check will be enforced. Otherwises, a decimal rescaling will be performed according to scale of columnVector and inputted roundingMode. 
Corresponding test codes are [here](https://github.com/rapidsai/cudf/pull/6609/files#diff-e7430fdc52346d6e5fafaf008865f29e50d7e3ee163e09b93cc1fc1a9c8d1ab6R83).